frontend/src/metabase/dashboard/components/DashboardGrid.jsx

```diff
@@ -96,7 +96,7 @@ class DashboardGrid extends Component {
   }
 
   UNSAFE_componentWillReceiveProps(nextProps) {
-    const { dashboard, dashcardData, isEditing } = nextProps;
+    const { dashboard, dashcardData, isEditing, selectedTabId } = nextProps;
```

Comment:
Is there a reason we are bringing the tab logic into DashboardGrid besides the fact that DashboardGrid is the connecting component between the `PublicDashboard` and `Dashboard`? It feels odd to be mixing the logic of
1. setting up the grid and organizing cards on the grid

      with
2. making the grid 'tab' aware by filtering cards based on the tab

It seems like this could complicate maintaining and extending either features in the future. Thoughts?

Reply:
This isn't a new pattern, @alxnddr also did this in https://github.com/metabase/metabase/pull/30630, where he added methods like [`getVisibleCards.`](https://github.com/metabase/metabase/blob/6709a06c9c4b5792165c4d9655fbe1ad62209421/frontend/src/metabase/dashboard/components/DashboardGrid.jsx#L179)

`DashboardGrid` isn't soley responsible for setting up the grid and organizing cards on it, the component that has the sole responsibility for arranging the grid is `GridLayout`, which is a wrapper over the `react-grid-layout` library. `DashboardGrid` contains higher level logic, with methods like `getVisibleCards` and `renderAddSeriesModal`. 

Logic like filtering cards on the current tab, as well as hiding cards that are empty as in Sasha's PR, should belong in `DashboardGrid`, because there are many components that render it for use in different contexts: `Dashboard` (in-app), `PublicDashboard` (for embedding/public sharing), `AutomaticDashboardApp` (for x-rays), `AuditDashboard` (for enterprise-edition). If we didn't implement these features in `DashboardGrid`, we would have to duplicate the logic across those four use cases (plus any new environments added in the future) and maintain the same logic across all of them.

Reply:
Thank you for the context, Emmad!